src/components/Theme/src/utils.js

```diff
@@ -48,24 +48,76 @@ function capitalizeFirstLetter(string) {
 export function resolveThemeableProps(componentKeyInTheme, propNames) {
 	const computedResolvedProps = {};
 	for (const propName of propNames) {
-		computedResolvedProps[`resolved${capitalizeFirstLetter(propName)}`] = function resolveThemeableProp() {
-			if (!isNil(this[propName])) {
-				// if validator for this prop exists then Vue would have already validated it by this point
-				return this[propName];
-			}
-			const themeDefaultPropValue = this.theme[componentKeyInTheme][propName];
-			if (themeDefaultPropValue) {
-				const valueOrPath = themeDefaultPropValue;
-				const value = this.theme.resolve(valueOrPath);
+		if (propName === 'variant') {
+			computedResolvedProps.resolvedVariant = function resolveThemeableVariant() {
+				// local override directly set on component instance
+				if (!isNil(this[propName])) {
+					// if validator for this prop exists then
+					// Vue would have already validated it by this point
+					return this[propName];
+				}
+
+				let valueOrPath;
+
+				// more specific theme override
+				const themeValue = this.theme[componentKeyInTheme][propName];
+				if (themeValue) {
+					valueOrPath = themeValue;
+				}
+
+				if (!valueOrPath) {
+					return undefined;
+				}
+
+				const resolvedValue = this.theme.resolve(valueOrPath);
+				const propValidator = this.$vnode.componentOptions
+					.Ctor.extendOptions.props[propName].validator;
+				if (propValidator) {
+					assert.error(propValidator(resolvedValue), `Invalid value "${resolvedValue}" for prop "${propName}" for component "${componentKeyInTheme}" in theme.`);
+				}
+				return resolvedValue;
+			};
+		} else {
+			computedResolvedProps[`resolved${capitalizeFirstLetter(propName)}`] = function resolveThemeableProp() {
+				// local override directly set on component instance
+				if (!isNil(this[propName])) {
+					// if validator for this prop exists then
+					// Vue would have already validated it by this point
+					return this[propName];
+				}
+
+				let valueOrPath;
+
+				// default theme value
+				const themeValue = this.theme[componentKeyInTheme][propName];
+				if (themeValue) {
+					valueOrPath = themeValue;
+				}
+
+				// variant value
+				const hasUserDefinedVariants = !this.$vnode.componentOptions
+					.Ctor.extendOptions.props.variant?.validator;
+				if (this.resolvedVariant && hasUserDefinedVariants) {
```

Comment:
This part I don't understand- It's my understanding that the `validator` is a method that should be called. It looks here like we're just checking for the existence of the method?  Why is that

Reply:
this is an internal maker implementation detail, it doesn't really affect the public api, but if you're curious the reason it was written this way is because some components resolve variants internally and others resolve it from the theme, and the validation method can be different for each scenario, i updated this code in my most recent commit anyway